### DISCUSSION TITLE
Is there a way that I could register a component after it has been mounted on the Canvas?

### DISCUSSION THREAD

**Discussion by saranatour1** [GrapesJS/grapesjs#5534] [Q&A]
I am working with React 18.2.0 with grapesJS 0.21.7, and I am using the base React code to mount my custom components, 
I was wondering if there's a way to register the components as a grapesjJS component after it has been mounted on the canvas, or even before that, like for instance reading the file and returning a file that has registered the components?   
also I was wondering if there's a way to group the traits in the trait manager, like if I have multiple set of slides, I want to see them on the side bar as traits (but grouped based on the trait type, or think of it as an array of traits)..
I have not been successful in my trials so I could use the help!

**Comment by artf** [comment]
> I was wondering if there's a way to register the components as a grapesjJS component after it has been mounted on the canvas, or even before that, like for instance reading the file and returning a file that has registered the components?

Sorry, but I'm not getting what you're trying to achieve.

> also I was wondering if there's a way to group the traits in the trait manager, like if I have multiple set of slides, I want to see them on the side bar as traits (but grouped based on the trait type, or think of it as an array of traits)..

Unfortunately [still waiting for help](https://github.com/GrapesJS/grapesjs/issues/1806)

**Comment by saranatour1** [comment]
> Sorry, but I'm not getting what you're trying to achieve.     
   
what I was meaning to say, is that in the base-react code, each parent component has it's own children that are also added within the same component registry, which while I was working on that approach, it made realize that this connot be effective for components that have alot of actions or toggles going within them, so I tried an approach where I just add the Parent component, and get the iframe body element, and map its children nodes until there's nothing left, and I wasn't able to continue with this approach, because I couldn't figure out how to make the Html recognizable after it has been on the canvas.. which now I realize this should be before all of that happens and not afterwards.    

I have seen the category issue as well, it's a problem that will save me alot of work and effort if it was done correctly, I'll see if I can do it some time soon, thanks alot @artf.
